frontend/src/scenes/data-warehouse/editor/draftsLogic.tsx

```diff
@@ -0,0 +1,158 @@
+        },
+        loadMoreDraftsSuccess: ({ draftsResponse }) => {
+            actions.setDrafts(draftsResponse.results)
+        },
+        saveAsDraft: async ({ query, viewId, successCallback, editedHistoryId }) => {
+            const draft = await api.dataWarehouseSavedQueryDrafts.create({
+                query,
+                saved_query_id: viewId,
+                edited_history_id: editedHistoryId,
+            })
+            lemonToast.success('Draft saved')
+            successCallback && successCallback(draft)
+
+            const newDrafts = [...values.drafts, draft]
+            actions.setDrafts(newDrafts)
+        },
+        updateDraft: async ({ draft }) => {
+            await api.dataWarehouseSavedQueryDrafts.update(draft.id, draft)
+            lemonToast.success('Draft updated')
+            const newDrafts = values.drafts.map((d) => (d.id === draft.id ? draft : d))
+            actions.setDrafts(newDrafts)
+        },
+        deleteDraft: async ({ draftId, successCallback }) => {
+            try {
+                await api.dataWarehouseSavedQueryDrafts.delete(draftId)
+                lemonToast.success('Draft deleted')
+                successCallback && successCallback()
+
+                const newDrafts = values.drafts.filter((draft) => draft.id !== draftId)
+                actions.setDrafts(newDrafts)
+            } catch (e) {
+                lemonToast.error('Failed to delete draft')
+                posthog.captureException(e)
+            }
+        },
+        renameDraft: async ({ draftId, name }) => {
+            await api.dataWarehouseSavedQueryDrafts.update(draftId, { name })
+            actions.setDrafts(values.drafts.map((d) => (d.id === draftId ? { ...d, name } : d)))
+        },
+        saveOrUpdateDraft: async ({
+            query,
+            viewId,
+            draftId,
+            editedHistoryId,
+        }: {
+            query: HogQLQuery
+            viewId?: string
+            draftId?: string
+            editedHistoryId?: string
+        }) => {
+            try {
+                if (draftId) {
+                    await api.dataWarehouseSavedQueryDrafts.update(draftId, {
+                        query,
+                    })
+                    lemonToast.success('Draft updated')
+                    const newDrafts = values.drafts.map((d) => (d.id === draftId ? { ...d, query } : d))
+                    actions.setDrafts(newDrafts)
+                } else {
+                    const existingDrafts = await api.dataWarehouseSavedQueryDrafts.list()
+                    const existingDraft = existingDrafts.results.find((draft) => draft.saved_query_id === viewId)
+
+                    if (existingDraft) {
+                        actions.updateDraft({
+                            ...existingDraft,
+                            query,
+                        })
+                        lemonToast.success('Draft updated')
```

Comment:
`actions.updateDraft` already shows a success toast, so we'll wanna remove this one to not double toast